srcDoc/custom_types.rst

```diff
@@ -0,0 +1,578 @@
+
+    struct Test {
+        long test_long;
+        TestEnum test_enum;
+        StringTest test_string;
+        SeqTest test_seq;
+    };//@Extensibility FINAL_EXTENSIBILITY
+
+These functions in the example are spreaded across several `idl` files to show the
+multiple files capability of the feature.
+
+These are the steps needed to use the above type in *RTI Perftest* for the
+C++ (Traditional) API:
+
+1. Copy all the `idl` files into the `srcIdl/customType/` folder.
+
+2. The following functions should be implemented (optionally) to properly
+   initialize and set the custom type structures.
+
+    - **initialize_custom_type_data**
+    This function is used to initialize your data.
+    Using this function, you will be able to allocate memory or set an immutable
+    field of the data.
+    The function takes one argument:
+
+        - A reference to custom type data.
+
+    .. code-block:: c
+
+        bool initialize_custom_type_data(RTI_CUSTOM_TYPE &data)
+        {
+            bool success = true;
+            if (!data.test_seq.test_seq.ensure_length(SIZE_TEST_SEQ, SIZE_TEST_SEQ)) {
+                success = false;
+            }
+            data.test_enum = ENUM1;
+            return success;
+        }
+
+    - **register_custom_type_data**
+    This function is used to set your data before being registered. It is only
+    required for key types. Set the key field of the data based on the key input.
+    There is a one-to-one mapping between an input key
+    and an instance.
+    The function takes two arguments:
+
+        - A reference to custom type data.
+        - A specific number unique for every key.
+
+    .. code-block:: c
+
+        void register_custom_type_data(RTI_CUSTOM_TYPE & data, unsigned long key)
+        {
+            data.test_long = key;
+        }
+
+    - **set_custom_type_data**
+    This function is used to set your data before it is sent.
+    It is called every time the data is sent.
+    You must set the custom type data before it is sent with the right
+    "key" value and the "targetDataLen".
+    The function takes three arguments:
+
+        - A reference to custom type data.
+        - A specific number unique for every key.
+        - The target size set by the command-line parameter ``-dataLen <bytes>``
+        minus the overhead of *RTI Perftest*. If applicable, you can use this
+        value to set the content of the data.
+
+    .. code-block:: c
+
+        bool set_custom_type_data(
+                RTI_CUSTOM_TYPE & data,
+                unsigned long key,
+                int targetDataLen)
+        {
+            bool success = true;
+            data.test_long = key;
+            if (sprintf(data.test_string.test_string, "Hello World! %lu", key) < 0) {
+                success = false;
+            }
+            return success;
+        }
+
+    - **finalize_custom_type_data**:
+    This function is used to remove your data. It is called in the destructor.
```

Comment:
I'm actually unsure why this finalize is needed?  Only if you're running some sort of memory checker/validation software for memory leaks?

Yes, it is programming "correctness" to have such a function.  But practically, speaking, there's no reason to implement this.  Having it available is mostly just wasted effort (by us to support/document it) other than satisfying those folks who are OCD...